plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -243,15 +232,46 @@ fun reauthConnectionIfNeeded(
     project: Project?,
     connection: ToolkitConnection,
     onPendingToken: (InteractiveBearerTokenProvider) -> Unit = {},
+    isReAuth: Boolean = false
```

Comment:
flip the condition because it's only false for the initial login case

Reply:
discussed offline, will continue to keep this as false

Reply:
we should only have isReAuth:true if user is re-logging/auth is expired. existing connection should be counted as first login only.

Reply:
Yes, this is an implementation detail.